### PR TITLE
Add support for time period unit.

### PR DESCRIPTION
**Pull request by iilieva**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Examples have been added / updated (for bug fixes / features)
-   [x] Changelog has been updated

## PR Type

What kind of change does this PR introduce?

-   [x] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Documentation content changes
-   [ ] Example website changes
-   [ ] Version bump
-   [ ] Other... Please describe:

## What does this change do?

- Introduced new unit for time period.
- Fixed bug in input with unit component - the unit text should be localized.
- Enahced some ts doc texts.

## What manual testing did you do?

Example app updated with time period control example functionality.

## Screenshots (if applicable)

Example App - Input with Time Period unit

Example App - Input with Time Period unit - read only view

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/examples/src/components/form-input/number-with-unit-form-input.example.component.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.form-value-display {
```
Comment (juanmendes) [inline projects/examples/src/components/form-input/number-with-unit-form-input.example.component.scss:1]: Why did we need these? Doesn't this mean the reusable component is not properly aligned? Shouldn't we fix it from the component itself?

File: projects/components/src/utils/unit/unit.ts [hunk 2]
```diff
@@ -214,3 +216,81 @@ export class Percent extends Unit {
...
+        TimePeriod.MONTHS,
+        TimePeriod.YEARS,
+    ];
+
+    static unitNameTranslationKeyPrefix = 'vcd.cc.units.timePeriod.';
+    static valueWithUnitTranslationKeyPrefix = 'vcd.cc.display.timePeriod.';
+
+    threshold: number;
+
+    constructor(multiplier: number, unitName: string, threshold: number) {
```
Comment (juanmendes) [inline projects/components/src/utils/unit/unit.ts:262]: Can you explain what threshold means in the documentation?

File: projects/components/src/utils/unit/unit.ts [hunk 3]
```diff
@@ -214,3 +216,81 @@ export class Percent extends Unit {
         return Percent.ZERO_TO_100;
     }
 }
+
+const SECONDS_IN_MINUTE: number = 60;
+const SECONDS_IN_HOUR: number = 3600;
```
Comment (juanmendes) [inline projects/components/src/utils/unit/unit.ts:221]: Like you did in `SECONDS_IN_DAY`
```ts
const SECONDS_IN_HOUR = SECONDS_IN_MINUTE * MINUTES_IN_HOUR ;

**Comment by juanmendes** [comment]
Please be sure to modify [CHANGELOG](https://github.com/vmware/vmware-cloud-director-ui-components/blob/master/projects/components/CHANGELOG.MD) explaining the changes. To publish a new version, bump the version in [components/package.json](https://github.com/vmware/vmware-cloud-director-ui-components/blob/master/projects/components/package.json) and the pipeline will do it automatically

File: projects/examples/src/components/form-input/number-with-unit-form-input.example.component.scss (hunk 1 above)
Comment (iilieva) [reply projects/examples/src/components/form-input/number-with-unit-form-input.example.component.scss:1]: This is just cosmetic change for the example view - not to show the selected values too close under the inputs. It seemed to me better to have a little distance between them. 
There is no relation to the form itself.

File: projects/components/src/utils/unit/unit.ts (hunk 2 above)
Comment (iilieva) [reply projects/components/src/utils/unit/unit.ts:262]: Done.

File: projects/components/src/utils/unit/unit.ts (hunk 3 above)
Comment (iilieva) [reply projects/components/src/utils/unit/unit.ts:221]: Done

**Pull request merged**